File: Neps/jaspion.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
    int t; cin >> t;
    int m, n;
    string jap, por, linha, erro;


    while(t--){
        map<string, string> dic;
        cin >> m >> n;
        for(int i = 0; i < m*2; i++){
            if(i % 2 != 0)
                getline(cin, jap);
            else{
                getline(cin, por);
                dic[jap] = por;
            }
        }
        while(n--){
            getline(cin, linha);
            string tmp="";
            for(int i = 0; i < int(linha.size()); i++){
                if(linha[i] == ' '){
                    if(dic.find(tmp) != dic.end()){
                        cout << dic[tmp] << ' ';
                    }else cout << tmp << ' ';
                    tmp = "";
                }
                else{
                    tmp += linha[i];
                }
            }
            cout << endl;
        }
    }
}

File: Neps/jaspion.py
def main():
    t = int(input())
    ab = input()

    for i in range(0, t):
        m = int(ab.split()[0])
        n = int(ab.split()[1])
        dic = dict()
        for j in range(0, m):
            jap = input()
            por = input()

            dic[jap] = por

        for j in range(0, n):
            palavra = input()

            if palavra.find(" "):
                for p in palavra.split():
                    if p not in dic.keys():
                        print(p, end=" ")
                    else:
                        print(dic[p], end=" ")
            else:
                if palavra not in dic.keys():
                    print(palavra, end=" ")
                else:
                    print(dic[palavra], end=" ")        
            print("\n")
		print("\n\n")

if __name__ == "__main__":
    main()
